3rd-party/vaultpress.php

```diff
@@ -1,26 +1,36 @@
 <?php
 
+/**
+ * Notify user that VaultPress has been disabled. Hide VaultPress notice that requested attention.
+ *
+ * @since 5.8
+ */
 function jetpack_vaultpress_rewind_enabled_notice() {
-	$plugin_file = 'vaultpress/vaultpress.php';
-
-	deactivate_plugins( $plugin_file );
 	?>
 	<div class="notice notice-success vp-deactivated">
 		<h2 style="margin-bottom: 0.25em;"><?php _e( 'Jetpack is now handling your backups.', 'jetpack' ); ?></h2>
 		<p><?php _e( 'VaultPress is no longer needed and has been deactivated.', 'jetpack' ); ?></p>
 	</div>
+	<style>#vp-notice{display:none;}</style>
 	<?php
 }
 
-// If Rewind is enabled, then show a notification to disable VaultPress.
+/**
+ * If Rewind is enabled, remove its entry in sidebar, deactivate VaultPress, and show a notification.
+ *
+ * @since 5.8
+ */
 function jetpack_vaultpress_rewind_check() {
 	if ( Jetpack::is_active() &&
 		 Jetpack::is_plugin_active( 'vaultpress/vaultpress.php' ) &&
 		 Jetpack::is_rewind_enabled()
 		) {
+		remove_submenu_page( 'jetpack', 'vaultpress' );
+
+		deactivate_plugins( 'vaultpress/vaultpress.php' );
```

Comment:
Moving this to here is probably a bad idea.

We had intentionally left it in the notice function, as there may be pages that `admin_init` runs on ([such as `admin_ajax`](https://github.com/WordPress/WordPress/blob/36b3f1707cd2a3d48a54a9637f37e2273b1fe319/wp-admin/admin-ajax.php#L45)) that would otherwise deactivate the plugin without the notification explaining ever being shown.

Reply:
Cool, I'll make the change and will add a comment explaining why it's there.